org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ClasspathManagement.scala

```diff
@@ -493,5 +494,42 @@ trait ClasspathManagement extends HasLogger { self: ScalaProject =>
     errors.toSeq
   }
 
+  /** Make sure compiler plugins are binary compatible with the selected Scala installation.
+   *
+   *  Unlike regular dependencies, compiler plugins must be cross-compiled with the exact compiler
+   *  version used for building.
+   */
+  private def validateCompilerPlugins(): Seq[(Int, String, String)] = if (!isUsingCompatibilityMode()) Seq() else {
```

Comment:
They are indeed linked. The `-Xsource` flag is telling the type-checker to emit the same errors as in that major revision of the compiler. It has no bearing on the build compiler. Setting a Scala installation from the 2.10 series will automatically add `-Xsource`.

Probably that test should not be there, in fact, since compiler plugins may be incompatible even when running with a 2.11 compiler.